--- Mozilla-DOM-0.21/xs/DOM.xs.orig	2007-06-06 23:05:57.000000000 +0200
+++ Mozilla-DOM-0.21/xs/DOM.xs	2008-07-22 07:56:56.908756844 +0200
@@ -4351,39 +4351,6 @@ moz_dom_ComparePoint (nsrange, parent, o
     OUTPUT:
 	RETVAL
 
-## IntersectsNode(nsIDOMNode *n, PRBool *_retval)
-PRBool
-moz_dom_IntersectsNode (nsrange, n)
-	nsIDOMNSRange *nsrange;
-	nsIDOMNode * n;
-    PREINIT:
-	PRBool _retval;
-    CODE:
-	nsrange->IntersectsNode(n, &_retval);
-	RETVAL = _retval;
-    OUTPUT:
-	RETVAL
-
-## CompareNode(nsIDOMNode *n, PRUint16 *_retval)
-PRUint16
-moz_dom_CompareNode (nsrange, n)
-	nsIDOMNSRange *nsrange;
-	nsIDOMNode * n;
-    PREINIT:
-	PRUint16 _retval;
-    CODE:
-	nsrange->CompareNode(n, &_retval);
-	RETVAL = _retval;
-    OUTPUT:
-	RETVAL
-
-## NSDetach(void)
-void
-moz_dom_NSDetach (nsrange)
-	nsIDOMNSRange *nsrange;
-    CODE:
-	nsrange->NSDetach();
-
 # -----------------------------------------------------------------------------
 
 
--- Mozilla-DOM-0.21/Makefile.PL.orig	2007-06-06 22:40:06.000000000 +0200
+++ Mozilla-DOM-0.21/Makefile.PL	2008-07-22 19:20:43.000000000 +0200
@@ -17,6 +17,7 @@
 my $mozpkg = 'mozilla-xpcom';
 my $ffpkg  = 'firefox-xpcom';
 my $mffpkg = 'mozilla-firefox-xpcom';
+my $xulpkg = 'libxul';
 
 our %build_reqs = (
     'perl-ExtUtils-Depends'   => '0.205',
@@ -24,6 +25,7 @@
     $mozpkg                   => '1.7',
     $ffpkg                    => '1.0',
     $mffpkg                   => '1.0',
+    $xulpkg                   => '1.9',
 );
 
 unless (eval "use ExtUtils::Depends '$build_reqs{'perl-ExtUtils-Depends'}';"
@@ -42,9 +44,10 @@
 
 my %pkgcfg = ExtUtils::PkgConfig->find("$mozpkg >= " . $build_reqs{$mozpkg},
                                        "$ffpkg >= " . $build_reqs{$ffpkg},
-                                       "$mffpkg >= " . $build_reqs{$mffpkg});
+                                       "$mffpkg >= " . $build_reqs{$mffpkg},
+                                       "$xulpkg >= " . $build_reqs{$xulpkg});
 
-my ($pkg) = $pkgcfg{pkg} =~ /^(\S+-xpcom)/;
+my ($pkg) = $pkgcfg{pkg} =~ /^(\S+)/;
 
 mkdir 'build', 0777;
 
@@ -63,7 +66,7 @@
 # -I/usr/include/mozilla/string -I/usr/include/mozilla/nspr
 $mozdom->set_inc($pkgcfg{cflags}, '-I.', '-I./build',
                  "-I$incdir", "-I$incdir/dom", "-I$incdir/webbrwsr",
-                 "-I$incdir/content", "-I$incdir/docshell");
+                 "-I$incdir/content", "-I$incdir/docshell", "-I$incdir/necko", "-I$incdir/xpcom");
 # -L/usr/lib/mozilla -lxpcom -lplds4 -lplc4 -lnspr4 -ldl -lc -lpthread
 $mozdom->set_libs($pkgcfg{libs});
 
