--- Mozilla-DOM-0.23/xs/DOM.xs.orig	2009-10-09 23:34:46.000000000 +0200
+++ Mozilla-DOM-0.23/xs/DOM.xs	2011-03-12 11:15:25.387412942 +0100
@@ -6678,7 +6678,7 @@ moz_dom_GetScrollTop (nshtmlelement)
     PREINIT:
 	PRInt32 aScrollTop;
     CODE:
-	nshtmlelement->GetScrollTop(&aScrollTop);
+	((nsIDOMNSElement*)nshtmlelement)->GetScrollTop(&aScrollTop);
 	RETVAL = aScrollTop;
     OUTPUT:
 	RETVAL
@@ -6689,7 +6689,7 @@ moz_dom_SetScrollTop (nshtmlelement, aSc
 	nsIDOMNSHTMLElement *nshtmlelement;
 	PRInt32  aScrollTop;
     CODE:
-	nshtmlelement->SetScrollTop(aScrollTop);
+	((nsIDOMNSElement*)nshtmlelement)->SetScrollTop(aScrollTop);
 
 ## GetScrollLeft(PRInt32 *aScrollLeft)
 PRInt32
@@ -6698,7 +6698,7 @@ moz_dom_GetScrollLeft (nshtmlelement)
     PREINIT:
 	PRInt32 aScrollLeft;
     CODE:
-	nshtmlelement->GetScrollLeft(&aScrollLeft);
+	((nsIDOMNSElement*)nshtmlelement)->GetScrollLeft(&aScrollLeft);
 	RETVAL = aScrollLeft;
     OUTPUT:
 	RETVAL
@@ -6709,7 +6709,7 @@ moz_dom_SetScrollLeft (nshtmlelement, aS
 	nsIDOMNSHTMLElement *nshtmlelement;
 	PRInt32  aScrollLeft;
     CODE:
-	nshtmlelement->SetScrollLeft(aScrollLeft);
+	((nsIDOMNSElement*)nshtmlelement)->SetScrollLeft(aScrollLeft);
 
 ## GetScrollHeight(PRInt32 *aScrollHeight)
 PRInt32
@@ -6718,7 +6718,7 @@ moz_dom_GetScrollHeight (nshtmlelement)
     PREINIT:
 	PRInt32 aScrollHeight;
     CODE:
-	nshtmlelement->GetScrollHeight(&aScrollHeight);
+	((nsIDOMNSElement*)nshtmlelement)->GetScrollHeight(&aScrollHeight);
 	RETVAL = aScrollHeight;
     OUTPUT:
 	RETVAL
@@ -6730,7 +6730,7 @@ moz_dom_GetScrollWidth (nshtmlelement)
     PREINIT:
 	PRInt32 aScrollWidth;
     CODE:
-	nshtmlelement->GetScrollWidth(&aScrollWidth);
+	((nsIDOMNSElement*)nshtmlelement)->GetScrollWidth(&aScrollWidth);
 	RETVAL = aScrollWidth;
     OUTPUT:
 	RETVAL
@@ -6742,7 +6742,7 @@ moz_dom_GetClientHeight (nshtmlelement)
     PREINIT:
 	PRInt32 aClientHeight;
     CODE:
-	nshtmlelement->GetClientHeight(&aClientHeight);
+	((nsIDOMNSElement*)nshtmlelement)->GetClientHeight(&aClientHeight);
 	RETVAL = aClientHeight;
     OUTPUT:
 	RETVAL
@@ -6754,7 +6754,7 @@ moz_dom_GetClientWidth (nshtmlelement)
     PREINIT:
 	PRInt32 aClientWidth;
     CODE:
-	nshtmlelement->GetClientWidth(&aClientWidth);
+	((nsIDOMNSElement*)nshtmlelement)->GetClientWidth(&aClientWidth);
 	RETVAL = aClientWidth;
     OUTPUT:
 	RETVAL
--- Mozilla-DOM-0.23/mozilladom2perl.h.orig	2007-06-06 23:46:36.000000000 +0200
+++ Mozilla-DOM-0.23/mozilladom2perl.h	2011-03-12 11:24:13.839417132 +0100
@@ -79,6 +79,7 @@ extern "C" {
 #include "nsIDOMElement.h"
 #include "nsIDOMEntity.h"
 #include "nsIDOMEntityReference.h"
+#include "nsIDOMNSElement.h"
 #include "nsIDOMNSEvent.h"
 #include "nsIDOMEvent.h"
 #include "nsIDOMEventListener.h"
